waypoint/main.py

```diff
@@ -0,0 +1,63 @@
+import os
+from contextlib import asynccontextmanager
+
+from fastapi import FastAPI
+
+from shared.constants import PROJECT_VERSION
+from shared.log_config import get_logger
+from waypoint.routes import sse
+from waypoint.services.dependency_injection.container import Container
+
+logger = get_logger(__name__)
+
+
+@asynccontextmanager
+async def app_lifespan(_: FastAPI):
+    logger.info("Waypoint Service startup")
+
+    container = Container()
+    await container.init_resources()
+
+    container.wire(modules=[__name__, sse])
+
+    await container.nats_events_processor()
+
+    yield
+
+    logger.debug("Shutting down Waypoint service...")
+    await container.shutdown_resources()
+    logger.info("Waypoint Service shutdown")
+
+
+def create_app() -> FastAPI:
+    openapi_name = os.getenv("OPENAPI_NAME", "Waypoint Service")
+
+    application = FastAPI(
+        title=openapi_name,
+        description="""
+        Welcome to the OpenAPI interface for the Waypoint Service.
+        The Waypoint Service enables the ability to wait for a specific event to occur before continuing.
+        """,
+        version=PROJECT_VERSION,
+        lifespan=app_lifespan,
+    )
+
+    application.include_router(sse.router)
+
+    return application
+
+
+logger.info("Waypoint Service startup")
+
+app = create_app()
+
+
+# TODO - Improve health checks
```

Comment:
let's include this :-)
We can maybe just add a `check_connection` method to NatsEventProcessor. Something which pings the JetStreamContext, if possible. Can be left for later if complicated, but maybe worth including now

health_check from webhooks for inspiration:
```py

@app.get("/health")
@inject
async def health_check(
    acapy_events_processor: AcaPyEventsProcessor = Depends(
        Provide[Container.acapy_events_processor]
    ),
    sse_manager: SseManager = Depends(Provide[Container.sse_manager]),
    billing_manager: BillingManager = Depends(Provide[Container.billing_manager]),
):
    if (
        acapy_events_processor.are_tasks_running()
        and sse_manager.are_tasks_running()
        and billing_manager.are_tasks_running()
    ):
        return {"status": "healthy"}
    else:
        raise HTTPException(
            status_code=503, detail="One or more background tasks are not running."
        )

```